Tests/UpstreamIntegrationTests/UpstreamIntegrationTests.swift

```diff
@@ -117,6 +116,282 @@ class UpstreamIntegrationTests: TestBase {
+        {
+          "payload": [
+            {
+              "ttlSeconds" : 1800,
+              "scope" : "EdgeNetwork",
+              "hint" : "stringType"
+            }
+          ]
+        }
+        """#
+
+        assertEdgeResponseEvent(expectedJSON: expectedLocationHintJSON, eventSource: TestConstants.EventSource.LOCATION_HINT_RESULT, exactMatchPaths: ["payload[*].scope"])
+
+        let expectedStateStore1stJSON = #"""
+        {
+          "payload": [
+            {
+              "maxAge": 1,
+              "key": "stringType",
+              "value": "stringType"
+            },
+            {
+              "maxAge": 1,
+              "key": "stringType",
+              "value": "stringType"
+            }
+          ]
+        }
+        """#
+
+        assertEdgeResponseEvent(expectedJSON: expectedStateStore1stJSON, eventSource: TestConstants.EventSource.STATE_STORE)
+    }
+
+    // MARK: - Configuration tests
+    // Tests standard sendEvent with both XDM and data, where data is complex - many keys and
+    // different value types
+    func testSendEvent_withSetLocationHint_receivesExpectedEventHandles() {
+        // Setup
+        let locationHintNetworkRequest = NetworkRequest(urlString: "https://obumobile5.data.adobedc.net/ee/va6/v1/interact", httpMethod: .post)!
+        networkService.setExpectationForNetworkRequest(networkRequest: locationHintNetworkRequest, expectedCount: 1)
+
+        Edge.setLocationHint("va6")
+
+        let eventPayloadJSON = #"""
+        {
+          "xdm": {
+            "testString": "xdm"
+          },
+          "data": {
+            "testDataString": "stringValue"
+          }
+        }
+        """#
+
+        let xdm = getAnyCodableAndPayload(eventPayloadJSON, type: .xdm)!
+        let data = getAnyCodableAndPayload(eventPayloadJSON, type: .data)!
+
+        let experienceEvent = ExperienceEvent(xdm: xdm.payload, data: data.payload)
+
+        // Test
+        Edge.sendEvent(experienceEvent: experienceEvent)
+
+        // Verify
+        // MARK: Network response assertions
+        let matchedResponse = networkService.getResponsesFor(networkRequest: locationHintNetworkRequest, timeout: 5)
+        XCTAssertEqual(200, matchedResponse.first?.responseCode)
+
+        // MARK: Response Event assertions
+        // Only validate for the location hint relevant to Edge Network extension
+        let expectedLocationHintJSON = #"""
+        {
+          "payload": [
+            {
+              "ttlSeconds" : 1800,
+              "scope" : "EdgeNetwork",
+              "hint" : "va6"
+            }
+          ]
+        }
+        """#
+
+        assertEdgeResponseEvent(expectedJSON: expectedLocationHintJSON, eventSource: TestConstants.EventSource.LOCATION_HINT_RESULT, exactMatchPaths: ["payload[*].scope", "payload[*].hint"])
+
+        let expectedStateStore1stJSON = #"""
+        {
+          "payload": [
+            {
+              "maxAge": 1,
+              "key": "stringType",
```

Comment:
we should include exact matches on key value at least

Reply:
yes, let's do exact match on "payload[0].value" where the expected value is `\#(locationHint)`